### PR TITLE
QOL-8518 prepare for CKAN 2.9

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Split Paster commands from Click
- Extract shared logic for Pylons and Flask routing
- Replace Nosetests with Pytest and partially unify tests for CKAN 2.8 and 2.9

### CODE REVIEW

File: ckanext/s3filestore/tests/__init__.py [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+    """
+    if hasattr(response, 'status_code'):
+        return response.status_code
+    elif hasattr(response, 'status_int'):
+        return response.status_int
+    else:
+        raise Exception("No status code found on %s" % response)
 
-# moto AWS mock is started externally on port 5000
-endpoint_url = config.get('ckanext.s3filestore.host_name', 'http://localhost:5000')
```
Comment (duttonw) [inline ckanext/s3filestore/tests/__init__.py:7]: so this is not needed?

**Review by duttonw — APPROVED** [review]
quite a refactor here, well done.

File: ckanext/s3filestore/tests/__init__.py (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/s3filestore/tests/__init__.py:7]: The S3 config should all be managed by `ckanext.s3filestore.uploader`. The tests make use of it at selected points.

This snippet was mostly for the benefit of the `setup_package` function, which nosetests used but pytest will not.

**Comment by ThrawnCA** [comment]
> quite a refactor here, well done.

The CKAN 2.9 tests were actually failing already, but the workflow ignored the failures. So there was a fair bit of rework required to get them sorted out.

**Comment by duttonw** [comment]
found that coveralls references the old org which has archived their project
it should be 
https://coveralls.io/github/qld-gov-au/ckanext-s3filestore?branch=develop
once we re-hook in coveralls to it

might be a good time to reach out to https://pypi.org/project/ckanext-s3filestore/ owners and see who will take over either us or https://github.com/keitaroinc/ckanext-s3filestore (also annoying that we seem to not be able to diff between us and keitaroinc

**Pull request merged**